src/transformers/tokenization_utils.py

```diff
@@ -359,10 +360,25 @@ def __init__(self, **kwargs):
         self._added_tokens_decoder.update(kwargs.pop("added_tokens_decoder", {}))
         self._added_tokens_encoder: Dict[str, int] = {k.content: v for v, k in self._added_tokens_decoder.items()}
 
-        # 4 init the parent class
+        # 4. init the parent class
         super().__init__(**kwargs)
 
-        # 4. If some of the special tokens are not part of the vocab, we add them, at the end.
+        # 5. Set the behavior when encountering an OOV token id
+        # Note: this must be set before _add_tokens, because _add_tokens 
+        # relies on the behavior of the OOV token ids for some tokenizers.
+        self.oov_error = kwargs.pop("oov_error", "")
+        # TODO @ArthurZ in version 5
+        if self.oov_error == "":
+            warnings.warn(
+                "The `oov_error` argument will be required in future versions of `transformers` (v5.0+), "
+                "to better handle Out-Of-Vocabulary (OOV) tokens. By default, this argument will be set "
+                "to 'replace', which converts OOV token IDs to an empty string. The original behavior can "
+                "be maintained by setting `oov_error` to 'strict'.",
+                FutureWarning,
+            )
```

Comment:
let's rather use `logger.warning_once` here to make sure this only appears once. 
```suggestion
            logger.warning_once(
                "The `oov_error` argument is set to the default `'replace'`. It will default to `'strict'` in `transformers` (v4.42) "
            )
```